.github/workflows/sudo-python-tests.yml

```diff
@@ -0,0 +1,85 @@
+# Copyright 2022 The Magma Authors.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Sudo python tests
+
+on:
+  workflow_dispatch: null
+  workflow_run:
+    workflows:
+      - build-all
+    branches:
+      - master
+    types:
+      - completed
+
+env:
+  CACHE_KEY: magma-dev-vm
+  REMOTE_DOWNLOAD_OPTIMIZATION: false
+
+jobs:
+  sudo-python-tests:
+    if: github.repository_owner == 'magma' || github.event_name == 'workflow_dispatch'
+    runs-on: macos-12
+    env:
+      SHA: ${{ github.event.workflow_run.head_commit.id || github.sha }}
```

Comment:
Top of my head I would not see, how those two values should be different?
As far as I understand the documentation the event triggering the workflow should automatically pass on the respective SHA.
Could you give any example, where this is different from the default behaviour (when omitting this line)?

Reply:
Pushed